tests/src/test/java/org/eclipse/hono/tests/mqtt5/MqttTestBase.java

```diff
@@ -0,0 +1,284 @@
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.tests.mqtt5;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.eclipse.hono.tests.IntegrationTestSupport;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hivemq.client.mqtt.MqttClientSslConfig;
+import com.hivemq.client.mqtt.datatypes.MqttClientIdentifier;
+import com.hivemq.client.mqtt.lifecycle.MqttClientDisconnectedListener;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5AsyncClient;
+import com.hivemq.client.mqtt.mqtt5.Mqtt5Client;
+import com.hivemq.client.mqtt.mqtt5.message.auth.Mqtt5SimpleAuth;
+import com.hivemq.client.mqtt.mqtt5.message.connect.Mqtt5Connect;
+import com.hivemq.client.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAck;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.core.net.SelfSignedCertificate;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * Base class for MQTT adapter integration tests using MQTT 5.0.
+ *
+ */
+public abstract class MqttTestBase {
+
+    private static final Vertx vertx = Vertx.vertx();
+
+    private static TrustManagerFactory trustManagerFactory;
+    private static HostnameVerifier acceptAllHostnames;
+
+    /**
+     * A logger to be used by subclasses.
+     */
+    protected final Logger LOGGER = LoggerFactory.getLogger(getClass());
+
+    /**
+     * A helper accessing the AMQP 1.0 Messaging Network and
+     * for managing tenants/devices/credentials.
+     */
+    protected IntegrationTestSupport helper;
+    /**
+     * A client for publishing messages to the MQTT protocol adapter.
+     */
+    protected Mqtt5AsyncClient mqttClient;
+    /**
+     * Creates default MQTT client TLS options.
+     *
+     * @throws Exception if the trust manager factory cannot be created.
+     */
+    @BeforeAll
+    public static void init() throws Exception {
+
+        trustManagerFactory = new PemTrustOptions()
+                .addCertPath(IntegrationTestSupport.TRUST_STORE_PATH)
```

Comment:
When I try to execute the integration tests I get the following error regarding this line:
```
java.lang.NullPointerException: No null certificate accepted

	at java.base/java.util.Objects.requireNonNull(Objects.java:235)
	at io.vertx.core.net.PemTrustOptions.addCertPath(PemTrustOptions.java:130)
	at org.eclipse.hono.tests.mqtt5.MqttTestBase.init(MqttTestBase.java:85)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at io.vertx.junit5.VertxExtension.interceptBeforeAllMethod(VertxExtension.java:130)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

I am guessing you have tested the integration tests on your end, so this is most likely a problem on my end. Have you encountered this issue before?

Reply:
Yes, the integration tests work on my machine ;-)
Did you build all of Hono before running the integration tests? The (test) certificates need to be packaged and installed to your local Maven repository before running the tests ...

Reply:
Ok, I got the tests to run successfully on my machine now :)